app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java

```diff
@@ -24,15 +29,21 @@ public void onClick(View v) {
             }
         });
 
-        checkForPermission(Manifest.permission.RECEIVE_SMS);
-        checkForPermission(Manifest.permission.READ_SMS);
+        checkForPermission(NEEDED_PERMISSIONS);
     }
 
-    private void checkForPermission(String permission) {
-        final int permissionStatus = ContextCompat.checkSelfPermission(this, permission);
-        if (permissionStatus == PackageManager.PERMISSION_GRANTED) {
-            return;
+    private void checkForPermission(String[] permissions) {
+        ArrayList<String> permissionsToAsk = new ArrayList<String>();
```

Comment:
Use interfaces whenever you can - `List<String> permissionsToAsk = new ArrayList<String>();`